src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/PropagateCancellationTokensWhenPossible.cs

```diff
@@ -2,14 +2,22 @@
 
 using System.Collections.Immutable;
 using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Operations;
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Threading;
 
 namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
 {
     /// <summary>
     /// Async005: Propagate CancellationTokens When Possible
     /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
     public abstract class PropagateCancellationTokensWhenPossibleAnalyzer : DiagnosticAnalyzer
     {
         internal const string RuleId = "Async005";
```

Comment:
Please change the RuleID and category to something more reasonable. Potentially, this can go under category `Microsoft.Performance` and you can pick the next available ID under it's range (CA18XX)